code/modules/power/supermatter/supermatter.dm

```diff
@@ -538,7 +538,7 @@
 
 	//Makes em go mad and accumulate rads.
 	for(var/mob/living/carbon/human/l in view(src, HALLUCINATION_RANGE(power))) // If they can see it without mesons on.  Bad on them.
-		if(!istype(l.glasses, /obj/item/clothing/glasses/meson) && !HAS_TRAIT(l, TRAIT_MESON_VISION))
+		if(!istype(l.glasses, /obj/item/clothing/glasses/meson) && !HAS_TRAIT(l, TRAIT_MESON_VISION) && !l.get_int_organ(/obj/item/organ/internal/eyes/cybernetic/meson))
```

Comment:
I'd rather see the meson implant to give the owner the `TRAIT_MESON_VISION` trait when it gets inserted. And removed again when the implant gets removed. The `insert` and `remove` proc defined on `/obj/item/organ` are perfect for this, you can override those.

Removing a head and replacing it with another should also work just fine since replacing a limb will properly call the `remove` and `instert` procs on the organs in the limb. But just in case, test this as well please

Reply:
I went ahead and did just that, and tested it with a bunch of surgeries and head amputations/transplants. Everything is working as it should. Let me know what you think.